subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/PluginDependencyDeclarationIntegrationTest.groovy

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.dsl.GradleDsl
+
+class PluginDependencyDeclarationIntegrationTest extends AbstractIntegrationSpec {
+
+    static final String JS = 'org.jetbrains.kotlin.js'
+    static final String JVM = 'org.jetbrains.kotlin.jvm'
+    static final String SCRIPTING = 'org.jetbrains.kotlin.plugin.scripting'
+    static final String VERSION = '1.8.0'
+    static final String STRICT_VERSION = '[1.0, 2.0['
+
+    def "understands plugin dependency notations from version catalog"() {
+        when:
+        settingsFile << """
+            dependencyResolutionManagement {
+                versionCatalogs {
+                    libs {
+                        plugin('js', "$JS").version {
+                            strictly "$STRICT_VERSION"
+                            prefer "$VERSION"
+                        }
+                        plugin('jvm', "$JVM").version("$VERSION")
+                        plugin('scripting', "$SCRIPTING").version("$VERSION")
+                    }
+                }
+            }
+        """
+
+        buildScript("""
+import org.gradle.api.internal.artifacts.dependencies.*
+
+// buildscript
+buildscript {
+    ${mavenCentralRepository()}
+    dependencies {
+        classpath(libs.plugins.js)
+        classpath(libs.plugins.jvm)
+    }
+}
+
+plugins {
+    id 'java'
+    id 'jvm-test-suite'
+}
+
+// main dependency block
+configurations {
+    conf
+}
+
+dependencies {
+    conf(libs.plugins.js)
+    conf(libs.plugins.jvm)
+    conf(libs.plugins.scripting)
+}
+
+// test-suite dependencies
+testing {
+    suites {
+        test {
+            dependencies {
+                implementation(plugin(libs.plugins.js))
+                implementation(plugin(libs.plugins.jvm))
+                implementation(plugin(libs.plugins.scripting))
```

Comment:
:question: This still looks different than in the top-level `dependencies {}` block.
Can't we make the top-level `dependencies {}` block DSL the same as the test-suites one?

Reply:
Good catch, I forgot to add the required `plugin(Provider<? extends PluginDependency> provider)` to the `DependencyHandler`, that's done now.